Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -52,12 +53,33 @@
 
 using namespace amrex;
 
+void
+WarpX::DepositChargeDensity (WarpXParticleContainer& species, const bool local,
+                             const bool reset, const bool do_rz_volume_scaling)
+{
+    species.DepositCharge(rho_fp, local, reset, do_rz_volume_scaling);
```

Comment:
Note that PR #1811 adds a call to SynchRho after calling `species.DepositCharge` (and effectively sets local=false). As noted there, this allows the option of applying filtering to the charge density. Note that that PR removes the `local` option since it is always false.

Reply:
That PR has been idle since April. Is it expected to move forward?

Reply:
@RemiLehe since that's your PR

Reply:
I think that PR #1811 might still move forward. 
Also, if I understood correctly, the changes related to `DepositChargeDensity` and `ChargeDensityGridProcessing` in the current PR are independent from the other changes and are purely a refactoring of the charge deposition. Is that correct?

If so, I would suggest to revert the changes associated with `ChargeDensityGridProcessing` and `DepositChargeDensity` from the current PR, so as to avoid a conflict with #1811

Reply:
Thanks for reviewing @RemiLehe. The refactoring to create `DepositChargeDensity` and `ChargeDensityGridProcessing` is just to break out the steps involved in the field solve somewhat more clearly and should hopefully not interfere too much with the other PR. The part that definitely does create a conflict is the switch to using `rho_fp` for the lab frame solution. We need this though for the external solver that we use to work (the reason for the `installpoissonsolver` part) since the `rho_fp` multifabs can be accessed from Python.
So I see three possible routes to clear things up:
1. Have this PR wait for #1811 to be merged, then update it with those changes afterwards
2. We split this out into 2 PR's: one for the `mlmg.getGradSolution` stuff now, and one for the `installpoissonsolver` and refactor stuff after #1811 is merged
3. Merge this one as is and update #1811 with these changes - I know this is the option that creates the least work for us and the most work for you, but I think it might be the simplest since it would basically just remove the `ChargeDensityGridProcessing` function and replace calls to it with a call to `SyncRho();` as well as remove the `local` flag from `DepositCharge`. If you'd like I'm happy to make those changes in a follow-up PR that will close 1811 as well.

Please let me know which route you think is best or if there is a different approach that would be even better.

Also just for your awareness the refactoring of the charge deposition is useful for us since it allows us to get the charge density for a specific species deposited on the grid and saved in `rho_fp` at any point during the simulation and then access that charge density from Python. We sometimes use this as a diagnostic.